cmd/abc/abc.go

```diff
@@ -51,13 +52,17 @@ func main() {
 		syscall.SIGINT, syscall.SIGTERM)
 	defer done()
 
-	if err := realMain(ctx); err != nil {
+	if err := realMain(ctx, os.Args[1:], os.Stdout, os.Stderr); err != nil {
 		done()
 		fmt.Fprintln(os.Stderr, err.Error())
 		os.Exit(1)
 	}
 }
 
-func realMain(ctx context.Context) error {
-	return rootCmd().Run(ctx, os.Args[1:]) //nolint:wrapcheck
+// argsTrimmed should be os.Args[1:], or the equivalent for testing.
+func realMain(ctx context.Context, argsTrimmed []string, stdout, stderr io.Writer) error {
+	cmd := rootCmd()
+	cmd.SetStdout(stdout)
+	cmd.SetStderr(stderr)
+	return cmd.Run(ctx, argsTrimmed) //nolint:wrapcheck
 }
```

Comment:
I don't think you need to do this. In tests, you can ask the Command to create a pipe, which will override stdin, stdout, and stderr to buffers: https://github.com/abcxyz/pkg/blob/f75fec6a9ad6e1b5d6b5a8fbb494f28644ad18af/cli/command_test.go#L308-L309

Reply:
Fixed, this is cleaner, thanks.